pallets/cdd-offchain-worker/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+            // nominators whom cdd claim got expired. internally it called the staking pallet functionality.
+            // Here we specify the function to be called back on-chain in next block import.
+            let call =
+                Call::take_off_invalidate_nominators(block, invalid_nominators.clone(), signature);
+
+            // Now let's create an unsigned transaction out of this call and submit it to the pool.
+            // By default unsigned transactions are disallowed, so we need to whitelist this case
+            // by writing `UnsignedValidator`. Note that it's EXTREMELY important to carefully
+            // implement unsigned validation logic, as any mistakes can lead to opening DoS or spam
+            // attack vectors. See validation logic docs for more details.
+            let _ = T::SubmitUnsignedTransaction::submit_unsigned(call)
+                .map_err(|()| "Unable to submit unsigned transaction.")?;
+        }
+        Ok(())
+    }
+
+    fn is_unsigned_transaction_allowed(block_number: T::BlockNumber) -> bool {
+        // check whether the last extrinsic submission blockNumber + cooling interval should be
+        // greater than current block number or not.
+        // It is not recommended to hook unsigned txn in every block because it is a non-deterministic
+        // task we are not sure how much time it will take to process. So cooling period is recommended
+        // to provide the gap for the execution of non-deterministic task.
+        Self::last_extrinsic_submitted_at() + T::CoolingInterval::get() < block_number
+    }
+}
+
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+    type Call = Call<T>;
+
+    /// Validate unsigned call to this module.
+    ///
+    /// By default unsigned transactions are disallowed, but implementing the validator
+    /// here we make sure that some particular calls (the ones produced by offchain worker)
+    /// are being whitelisted and marked as valid.
+    fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
+        // Firstly let's check that we call the right function.
+        if let Call::take_off_invalidate_nominators(block_number, target, signature) = call {
+            // Now let's check if the transaction has any chance to succeed.
+            let last_unsigned_at = <LastExtSubmittedAt<T>>::get();
+            ensure!(
+                last_unsigned_at + T::CoolingInterval::get() <= *block_number,
+                InvalidTransaction::Stale
+            );
+
+            // Let's make sure to reject transactions from the future.
+            ensure!(
+                &<system::Module<T>>::block_number() >= block_number,
+                InvalidTransaction::Future
+            );
+
+            // Get the key from the store to verify whether the signed signature is
+            // generated from the validator or not.
+            let signer_id = T::SignerId::all()
```

Comment:
Where are these `SignerId` coming from? I don't see how the validator id is being set in this type/var.